Test Failure: PolicyExecutorVirtualThreadTest.testMaxConcurrencyWithVirtualThreads

Test Failure: com.ibm.ws.threading.policy.PolicyExecutorVirtualThreadTest.testMaxConcurrencyWithVirtualThreads

```
testMaxConcurrencyWithVirtualThreads:junit.framework.AssertionFailedError: 2025-03-16-04:38:58:708 ERROR: Caught exception attempting to call test method testMaxConcurrencyWithVirtualThreads on servlet web.vt.PolicyVirtualThreadServlet
java.lang.AssertionError: expected:<true> but was:<false>
    at web.vt.PolicyVirtualThreadServlet.testMaxConcurrencyWithVirtualThreads(PolicyVirtualThreadServlet.java:195)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at web.vt.PolicyVirtualThreadServlet.doGet(PolicyVirtualThreadServlet.java:89)
    at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:633)
    at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:723)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1266)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:754)
    at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:451)
    at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1362)
    at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5096)
    at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:328)
    at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1047)
    at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1284)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:500)
    at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:459)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:569)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:503)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:363)
    at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:330)
    at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
    at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:516)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:586)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:970)
    at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1059)
    at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:298)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.base/java.lang.Thread.run(Thread.java:1583)

    at componenttest.topology.utils.FATServletClient.assertTestResponse(FATServletClient.java:106)
    at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:91)
    at com.ibm.ws.threading.policy.PolicyExecutorVirtualThreadTest.testMaxConcurrencyWithVirtualThreads(PolicyExecutorVirtualThreadTest.java:58)
    at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:216)
    at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:381)
    at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:185)
```

Test is failing on:
`assertEquals(true, threadNum4 >= '1' && threadNum4 <= '4');`

Max concurrency of 3 and the locking/coordination of tasks by the test only guarantees the first 3 will have consecutive numbering.  After that, the first 3 tasks will be able to complete and depending on timing could each create a virtual thread to continue processing queued tasks.  The test case assertion needs to be updated accordingly.

--- a/dev/com.ibm.ws.threading_policy_fat/test-applications/vtpolicyapp/src/web/vt/PolicyVirtualThreadServlet.java
+++ b/dev/com.ibm.ws.threading_policy_fat/test-applications/vtpolicyapp/src/web/vt/PolicyVirtualThreadServlet.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2023,2024 IBM Corporation and others.
+ * Copyright (c) 2023,2025 IBM Corporation and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
@@ -189,10 +189,13 @@ public class PolicyVirtualThreadServlet extends HttpServlet {
         char threadNum3 = thread3.getName().charAt(thread3.getName().length() - 1);
         char threadNum4 = thread4.getName().charAt(thread4.getName().length() - 1);
 
-        assertEquals(true, threadNum1 >= '1' && threadNum1 <= '4');
-        assertEquals(true, threadNum2 >= '1' && threadNum2 <= '4');
-        assertEquals(true, threadNum3 >= '1' && threadNum3 <= '4');
-        assertEquals(true, threadNum4 >= '1' && threadNum4 <= '4');
+        assertEquals(true, threadNum1 >= '1' && threadNum1 <= '3');
+        assertEquals(true, threadNum2 >= '1' && threadNum2 <= '3');
+        assertEquals(true, threadNum3 >= '1' && threadNum3 <= '3');
+        // Each of the above threads can create another virtual thread for tasks
+        // that remain on the queue. Any of those threads could end up running the
+        // fourth task.
+        assertEquals(true, threadNum4 >= '4' && threadNum4 <= '6');
 
         executor.shutdownNow();
     }

